### PR TITLE
Ranged Pruning

### PR DESCRIPTION
**Pull request by mjohnson541**
This PR updates pruning to work properly with ranged reactors and solve #1386.  Instead of tracking how many times a ranged reactor is completed ranged reactors are simply run through nSimsTerm times each reactor cycle.  Furthermore, maxEdgeSpeciesRateRatios is now only reset for each reactor at the beginning of the reactor loop allowing ranged reactors to store the maxEdgeSpeciesRateRatios of all the runs they did that reactor cycle instead of just the last cycle.  This also solves the minimum number of iterations to pruning issues as the number of iterations is defined as the number of reactor cycles run.  I've also added an example for testing pruning and updated the documentation to reflect the changes to how ranged reactors operate.

### CODE REVIEW

**Comment by rwest** [comment]
Could we consider changing the name of the "nSimsTerm" input parameter, as we are changing it's meaning? I found the name not very clear intuitive or self-documenting when I first came across it. 
Maybe not worth it - just a suggestion.

**Comment by mjohnson541** [comment]
I agree, it breaks input back compatibility, but the feature hasn't been around that long so its probably worth it.

**Comment by mjohnson541** [comment]
Can we try to get this merged?  I'm starting to do stuff in other PR's that will create some really dangerous merge conflicts whenever this or the other one needs rebased.

**Comment by mliu49** [comment]
The test case I was using finished, and this branch does much better than master at keeping a reasonable edge. I think it's still smaller than I would normally expect with standard reactors, but it's definitely better than the behavior on master.

File: rmgpy/rmg/main.py [hunk 1]
```diff
@@ -801,9 +802,9 @@ def execute(self, **kwargs):
                                 logging.warn('Reaction thresholds/flags for Reaction System {0} was not updated due to resurrection'.format(index+1))
```
Comment (mliu49) [inline rmgpy/rmg/main.py]: Should this line also be indented? I think the arguments and closing parenthesis in the function calls above should also be indented.

File: rmgpy/rmg/main.py [hunk 2]
```diff
@@ -817,27 +818,23 @@ def execute(self, **kwargs):
...
-                    self.saveEverything()
+                        maxNumSpcsHit = len(self.reactionModel.core.species) >= modelSettings.maxNumSpecies
+    
+                        if maxNumSpcsHit: #breaks the while loop 
+                            break
+                        
+                        if not reactorDone:
+                            self.done = False
+                        
+
```
Comment (mliu49) [inline rmgpy/rmg/main.py]: Could you only leave one empty line?

File: rmgpy/rmg/main.py [hunk 3]
```diff
@@ -688,111 +687,113 @@ def execute(self, **kwargs):
                 
                 for index, reactionSystem in enumerate(self.reactionSystems):
                     
-                    reactionSystem.prunableSpecies = prunableSpecies
+                    reactionSystem.prunableSpecies = prunableSpecies   #these lines reset pruning for a new cycle
                     reactionSystem.prunableNetworks = prunableNetworks
+                    reactionSystem.reset_max_edge_species_rate_ratios()
```
Comment (mliu49) [inline rmgpy/rmg/main.py]: Why is this needed in the new setup?

Comment (mjohnson541) [reply rmgpy/rmg/main.py]: This resets the maxEdgeSpeciesRateRatios array to zero, this isn't done in simulate anymore since we need to accumulate the values across all ranged reactor simulations.

File: rmgpy/rmg/main.py (hunk 1 above)
Comment (mjohnson541) [reply rmgpy/rmg/main.py]: done

File: rmgpy/rmg/main.py (hunk 2 above)
Comment (mjohnson541) [reply rmgpy/rmg/main.py]: done

File: rmgpy/rmg/main.py [hunk 4]
```diff
@@ -688,122 +687,124 @@ def execute(self, **kwargs):
...
                                 )
-                            else:
-                                self.updateReactionThresholdAndReactFlags(
+                                else:
+                                    self.updateReactionThresholdAndReactFlags(
                                         rxnSysUnimolecularThreshold = reactionSystem.unimolecularThreshold,
                                         rxnSysBimolecularThreshold = reactionSystem.bimolecularThreshold,
                                         rxnSysTrimolecularThreshold = reactionSystem.trimolecularThreshold,
                                         skipUpdate = True
                                 )
```
Comment (mliu49) [inline rmgpy/rmg/main.py]: Could you fix the indentation of this line and lines 791-794?

**Comment by mliu49** [comment]
I think you can also consider adding #1407 to this PR since it's only one commit.

**Comment by mjohnson541** [comment]
Done

**Comment by mjohnson541** [comment]
Is this ready to merge?
